### PR TITLE
Sodiq

### PR DESCRIPTION
**Pull request by mightysodiq**
implemented profile update

### CODE REVIEW

File: EatGoodNaija.Server/Model/DTO/profileDTO/UserProfileUpdateDTO.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿using Org.BouncyCastle.Asn1.Crmf;
+using System.ComponentModel.DataAnnotations;
+
+namespace EatGoodNaija.Server.Model.DTO.profileDTO
+{
+    public class UserProfileUpdateDTO
```
Comment (rocklessg) [inline EatGoodNaija.Server/Model/DTO/profileDTO/UserProfileUpdateDTO.cs]: You're repeating code here (violating DRY). You can make this class inherit from UserProfileCreateDTO. E.g

public class UserProfileUpdateDTO : UserProfileCreateDTO
{
}

File: EatGoodNaija.Server/Services/Implementation/UserRepository.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                HomeAddress = userRequest.HomeAddress,
+                City = userRequest.City
+            };
+
+            _context.UserProfile.Add(newUserProfile);
+            await _context.SaveChangesAsync();
+
+            return "User profile added successfully";
+        }
+
```
Comment (rocklessg) [inline EatGoodNaija.Server/Services/Implementation/UserRepository.cs]: Close white spaces

File: EatGoodNaija.Server/Controllers/ProfileController.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+            var existingUser = await _userRepository.UpdateProfile (profileUpdateDTO,userId);
+            if (existingUser == null)
+            {
+                return BadRequest();
+            }
+
+            return Ok(existingUser);
+        }
+
```
Comment (rocklessg) [inline EatGoodNaija.Server/Controllers/ProfileController.cs]: Close white spaces

File: EatGoodNaija.Server/Data/DataContext.cs [hunk 4]
```diff
@@ -21,6 +21,8 @@ public class DataContext : IdentityDbContext<Vendor>
         public DbSet<FoodItem> FoodItems { get; set; }
         public DbSet<ConfirmEmailToken> ConfirmEmailTokens { get; set; }
 
+        public DbSet<UserProfile> UserProfile { get; set; }
```
Comment (rocklessg) [inline EatGoodNaija.Server/Data/DataContext.cs]: Did you include this model (UserProfile) to your database tables ?

File: EatGoodNaija.Server/Controllers/ProfileController.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        public async Task<IActionResult> UpdateUserProfile(UserProfileUpdateDTO profileUpdateDTO, int userId)
+        {
+
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            var existingUser = await _userRepository.UpdateProfile (profileUpdateDTO,userId);
+            if (existingUser == null)
```
Comment (rocklessg) [inline EatGoodNaija.Server/Controllers/ProfileController.cs]: This check should be done in your service layer not in your controller.

File: EatGoodNaija.Server/Controllers/ProfileController.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+
+        [HttpPost]
+        public async Task<IActionResult> CreateUserProfile(UserProfileCreateDTO profileCreateDTO)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
```
Comment (rocklessg) [inline EatGoodNaija.Server/Controllers/ProfileController.cs]: Remove white spaces

File: EatGoodNaija.Server/Services/Interface/IUserRepository.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+using EatGoodNaija.Server.Model.DTO;
+using EatGoodNaija.Server.Model.DTO.profileDTO;
+namespace EatGoodNaija.Server.Services.Interface
+{
+    public interface IUserRepository
+    {
+        Task<UserProfile> GetById(int userId);
+        Task<string> UpdateProfile(UserProfileUpdateDTO user, int UserId);
+        Task<string> Add(UserProfileCreateDTO userRequest);
+
```
Comment (rocklessg) [inline EatGoodNaija.Server/Services/Interface/IUserRepository.cs]: Remove white spaces

File: EatGoodNaija.Server/Controllers/ProfileController.cs [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        public ProfileController(IUserRepository userRepository)
+        {
+            _userRepository = userRepository;
+        }
+
+        [HttpGet("{userId}")]
+        public async Task<ActionResult<UserProfile>>  GetUserProfile(int userId)
+        {
+            var user = await _userRepository.GetById(userId);
+            if (user == null)
```
Comment (rocklessg) [inline EatGoodNaija.Server/Controllers/ProfileController.cs]: This check should be done in your service layer

**Review by rocklessg — COMMENTED** [review]
- Pay attention to closing/Removing white spaces in your code.
- Respond and effect the comments.

File: EatGoodNaija.Server/Model/DTO/profileDTO/UserProfileUpdateDTO.cs (hunk 1 above)
Comment (mightysodiq) [reply EatGoodNaija.Server/Model/DTO/profileDTO/UserProfileUpdateDTO.cs]: implemented sir

File: EatGoodNaija.Server/Services/Implementation/UserRepository.cs (hunk 2 above)
Comment (mightysodiq) [reply EatGoodNaija.Server/Services/Implementation/UserRepository.cs]: implemented sir

File: EatGoodNaija.Server/Services/Implementation/datacontext.cs [hunk 9]
```diff
@@ -0,0 +1,6 @@
+﻿namespace EatGoodNaija.Server.Services.Implementation
+{
+    internal class datacontext
```
Comment (rocklessg) [inline EatGoodNaija.Server/Services/Implementation/datacontext.cs]: What is this class doing?

File: EatGoodNaija.Server/Migrations/20240423203620_dav2.cs [hunk 10]
```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace EatGoodNaija.Server.Migrations
+{
+    public partial class dav2 : Migration
```
Comment (rocklessg) [inline EatGoodNaija.Server/Migrations/20240423203620_dav2.cs:7]: Your migration name should be descriptive. E.g 
- AddedProductNameColumnToTheProductTable
   OR
- ChangedIntegerPhoneNumberColumnToStringType

File: EatGoodNaija.Server/Migrations/20240418172125_rt111.cs [hunk 11]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace EatGoodNaija.Server.Migrations
+{
+    public partial class rt111 : Migration
```
Comment (rocklessg) [inline EatGoodNaija.Server/Migrations/20240418172125_rt111.cs]: Nothing is added to this migration. What purpose is it serving in the application ?

File: EatGoodNaija.Server/Migrations/20240416063053_r1.cs [hunk 12]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace EatGoodNaija.Server.Migrations
+{
+    public partial class r1 : Migration
```
Comment (rocklessg) [inline EatGoodNaija.Server/Migrations/20240416063053_r1.cs]: Nothing is added to this migration. What purpose is it serving in the application ?

File: EatGoodNaija.Server/Migrations/20240416061951_w12.cs [hunk 13]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace EatGoodNaija.Server.Migrations
+{
+    public partial class w12 : Migration
+    {
```
Comment (rocklessg) [inline EatGoodNaija.Server/Migrations/20240416061951_w12.cs]: Nothing is added to this migration. What purpose is it serving in the application ?

File: EatGoodNaija.Server/Migrations/20240416060657_w3.cs [hunk 14]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace EatGoodNaija.Server.Migrations
+{
+    public partial class w3 : Migration
```
Comment (rocklessg) [inline EatGoodNaija.Server/Migrations/20240416060657_w3.cs]: Nothing is added to this migration. What purpose is it serving in the application?

File: EatGoodNaija.Server/Migrations/20240416055533_w1.cs [hunk 15]
```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace EatGoodNaija.Server.Migrations
+{
+    public partial class w1 : Migration
+    {
```
Comment (rocklessg) [inline EatGoodNaija.Server/Migrations/20240416055533_w1.cs]: Nothing is added to this migration. What purpose is it serving in the application?

File: EatGoodNaija.Server/Controllers/AuthController.cs [hunk 16]
```diff
@@ -108,5 +110,25 @@ public async Task<IActionResult> ConfirmEmail(ConfirmEmailTokenDTO token, string
                 return BadRequest(result);
             }
         }
+
+        [HttpPut("update-profile")]
+        public async Task<IActionResult> UpdateUserProfile(UpdateProfileDTO updateProfileDTO, string Id)
+        {
+            if (!ModelState.IsValid)
```
Comment (rocklessg) [inline EatGoodNaija.Server/Controllers/AuthController.cs]: Your controller should have minimum line of code possible. Do all checks in your services and or repository. E.g

public async Task<IActionResult> UpdateUserProfile(UpdateProfileDTO updateProfileDTO, string Id)
        {
            if (ModelState.IsValid)
           {
                var response= await _authService.UpdateProfileAsync(updateProfileDTO, Id);
                return Ok(response);
           }

**Review by rocklessg — COMMENTED** [review]
Remove all unused classes and migrations

File: EatGoodNaija.Server/Migrations/20240423203620_dav2.cs (hunk 10 above)
Comment (mightysodiq) [reply EatGoodNaija.Server/Migrations/20240423203620_dav2.cs:7]: noted!

**Pull request merged**